src/xml_parsing.cpp

```diff
@@ -709,32 +715,34 @@ TreeNode::Ptr XMLParser::PImpl::createNodeFromXML(const XMLElement* element,
                                     ") but not in the providedPorts() of its "
                                     "registered node type."));
         }
-        else
+
+        const auto& port_model = port_model_it->second;
+        bool is_blacbkboard = port_value.size() >= 3 && port_value.front() == '{' &&
+                              port_value.back() == '}';
+        // let's test already if conversion is possible
+        if(!is_blacbkboard && port_model.converter() && port_model.isStronglyTyped())
         {
-          const auto& port_model = port_model_it->second;
-          bool is_blacbkboard = port_value.size() >= 3 && port_value.front() == '{' &&
-                                port_value.back() == '}';
-          // let's test already if conversion is possible
-          if(!is_blacbkboard && port_model.converter() && port_model.isStronglyTyped())
+          // This may throw
+          try
           {
-            // This may throw
-            try
-            {
-              port_model.converter()(port_value);
-            }
-            catch(std::exception& ex)
-            {
-              auto msg = StrCat("The port with name \"", port_name, "\" and value \"",
-                                port_value, "\" can not be converted to ",
-                                port_model.typeName());
-              throw LogicError(msg);
-            }
+            port_model.converter()(port_value);
+          }
+          catch(std::exception& ex)
+          {
+            auto msg =
+                StrCat("The port with name \"", port_name, "\" and value \"", port_value,
+                       "\" can not be converted to ", port_model.typeName());
+            throw LogicError(msg);
           }
         }
       }
 
       port_remap[port_name] = port_value;
     }
+    else if(node_type == NodeType::SUBTREE && port_name == "_autoremap")
+    {
+      do_autoremap = convertFromString<bool>(port_value);
```

Comment:
Can we calculate this variable in its own loop or use the find attribute of the XML parser? It would be nice if this were its own code block.